### PR TITLE
[jaeger] unify all used tags, introduce builder pattern, additional…

### PR DESCRIPTION
**Pull request by drahnr**
… candidate annotations

Also moves the span name type from `impl Into<String>` to `&static str`, which will be required for #2277 .

Part of #2389

### CODE REVIEW

**Comment by ordian** [comment]
> additional tags for parentless/top-level candidate-spans to denote the stage in a numeric fashion stage:0 to stage:N would be my suggestion to simplify life for tooling or pass an explicit mapping of candidate stages mapped from existing span names

could you elaborate which tooling would it simplify? why not simply `{:?}, stage` instead of numbers?

**Comment by drahnr** [comment]
> > additional tags for parentless/top-level candidate-spans to denote the stage in a numeric fashion stage:0 to stage:N would be my suggestion to simplify life for tooling or pass an explicit mapping of candidate stages mapped from existing span names
> 
> could you elaborate which tooling would it simplify? why not simply `{:?}, stage` instead of numbers?

The numbers imply a clear ordering, as such the tooling does not need to be aware of possibly changing stages and new (meta-/sub-)stages can be added as needed.

File: node/jaeger/src/lib.rs [hunk 1]
```diff
@@ -157,6 +159,83 @@ impl std::ops::Deref for PerLeafSpan {
...
+/// statistical metrics for which stage of the inclusion
+/// pipeline drops a significant amount of candidates,
+/// statistically speaking.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+#[repr(u8)]
+#[non_exhaustive]
+pub enum Stage {
+	Backing = 1,
+	Availability = 2,
+	// TODO expand this
```
Comment (ordian) [inline node/jaeger/src/lib.rs:174]: approved?

**Review by ordian — APPROVED** [review]

File: node/jaeger/src/lib.rs (hunk 1 above)
Comment (drahnr) [reply node/jaeger/src/lib.rs:174]: This is really just the beginning of the stages. I wanted to add this PR so @insipx could possibly add sufficient stages to be unblocked.

**Comment by drahnr** [comment]
bot merge

**Review by eskimor — COMMENTED** [review]
I like :-)

File: node/jaeger/src/lib.rs (hunk 1 above)
Comment (rphmeier) [reply node/jaeger/src/lib.rs:174]: We try not to merge TODOs into the code unless they are tagged with an issue

Comment (drahnr) [reply node/jaeger/src/lib.rs:174]: Currently working on this :) so will be resolved early tomorrow morning.

Comment (drahnr) [reply node/jaeger/src/lib.rs:174]: Followup PR #2477

**Pull request merged**